### PR TITLE
fix(CRATemplate): Resolve React 18 dependency clash

### LINKED ISSUE #3213: Smoke test failing following React 18 release
`npx create-react-app@latest --template defencedigital myapp` is failing following the release of React 18 due to peer dependency errors:

```
Installing packages. This might take a couple of minutes.
Installing react, react-dom, and react-scripts with cra-template-defencedigital...


added 1370 packages in 39s

169 packages are looking for funding
  run `npm fund` for details

Initialized a git repository.

Installing template dependencies using npm...
npm ERR! code ERESOLVE
npm ERR! ERESOLVE could not resolve
npm ERR! 
npm ERR! While resolving: @apollo/client@3.5.10
npm ERR! Found: react@18.0.0
npm ERR! node_modules/react
npm ERR!   peer react@"^18.0.0" from react-dom@18.0.0
npm ERR!   node_modules/react-dom
npm ERR!     react-dom@"^18.0.0" from the root project
npm ERR!     peer react-dom@"*" from @testing-library/react@12.1.4
npm ERR!     node_modules/@testing-library/react
npm ERR!       @testing-library/react@"^12.1.2" from the root project
npm ERR!     1 more (styled-components)
npm ERR!   peer react@">= 16" from react-scripts@5.0.0
npm ERR!   node_modules/react-scripts
npm ERR!     react-scripts@"5.0.0" from the root project
npm ERR!   4 more (the root project, @testing-library/react, ...)
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peerOptional react@"^16.8.0 || ^17.0.0" from @apollo/client@3.5.10
npm ERR! node_modules/@apollo/client
npm ERR!   @apollo/client@"^3.5.6" from the root project
npm ERR! 
npm ERR! Conflicting peer dependency: react@17.0.2
npm ERR! node_modules/react
npm ERR!   peerOptional react@"^16.8.0 || ^17.0.0" from @apollo/client@3.5.10
npm ERR!   node_modules/@apollo/client
npm ERR!     @apollo/client@"^3.5.6" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
```

Blog post on React 18:

https://reactjs.org/blog/2022/03/29/react-v18.html

### PR DESCRIPTION
**Pull request by m7kvqbe1**
## Related issue

Closes #3213

## Overview

Fix conflicting React dependencies and add override for flagged vulnerability in deeply nested dep.

### CODE REVIEW

File: packages/cra-template-defencedigital/template.json [hunk 1]
```diff
@@ -18,7 +18,7 @@
       "@graphql-codegen/typescript-react-apollo": "^3.2.2",
       "@juggle/resize-observer": "^3.3.1",
       "@testing-library/jest-dom": "^5.5.0",
-      "@testing-library/react": "^12.1.2",
+      "@testing-library/react": "^13.1.1",
       "@testing-library/user-event": "^13.5.0",
       "@types/jest": "^27.0.3",
       "@types/node": "^16.11.13",
```
Comment (jpveooys) [inline packages/cra-template-defencedigital/template.json:24]: Would be good to update `@types/react` and `@types/react-dom` below to 18.x versions as well (to match the version of React installed)

**Review by jpveooys — COMMENTED** [review]
Nice!

File: packages/cra-template-defencedigital/template.json (hunk 1 above)
Comment (m7kvqbe1) [reply packages/cra-template-defencedigital/template.json:24]: Nice spot!

**Review by jpveooys — APPROVED** [review]
:+1:

**Pull request merged**